Added a reset handler sample (#85)
AxonIQ/code-samples: reset-handler/README.md

> We recommend checking the [FrameworkEventProcessorRestController.java](https://github.com/AxonIQ/code-samples/blob/master/reset-handler/src/main/java/io.axoniq/server/FrameworkEventProcessorRestController.java) for more information.
   
### Running the application
This is a Spring boot application, as such it can be ran as any other standard Spring Boot application. It has a simple `/event` endpoint where you can create new empty events. For resetting the token, it provides 2 reset endpoints:
- `/server/reset/{processorName}`
- `/framework/reset/{processorName}`

It also provides `/server/start/{processorName}` and `/server/pause/{processorName}` for testing purposes.
  
Since Axon Server is a requirement for this sample, a `docker-compose` file is provided.

Also, if you are on Intellij, a `requests.http` file is provided to make it easy to call the endpoints.

Most of the logic for the Axon Server reset is on the [EventProcessorService.java](https://github.com/AxonIQ/code-samples/blob/master/reset-handler/src/main/java/io.axoniq/server/EventProcessorService.java) class and the added javadoc should be enough to explain what it does.

> For the Axon Framework version, checking the official [StreamingEventProcessor.java](https://github.com/AxonFramework/AxonFramework/blob/master/messaging/src/main/java/org/axonframework/eventhandling/StreamingEventProcessor.java) is my best advice.

Reviewer: ```suggestion
For the Axon Framework version, we recommend checking the official [StreamingEventProcessor.java](https://github.com/AxonFramework/AxonFramework/blob/master/messaging/src/main/java/org/axonframework/eventhandling/StreamingEventProcessor.java) documentation.
```
